jax/scipy/special.py

```diff
@@ -101,15 +101,24 @@ def expit(x):
 
 @_wraps(osp_special.logsumexp)
 def logsumexp(a, axis=None, b=None, keepdims=False, return_sign=False):
-  if b is not None or return_sign:
-    raise NotImplementedError("Only implemented for b=None, return_sign=False")
   dims = _reduction_dims(a, axis)
   dimadd = lambda x: lax.expand_dims(x, dims)
   amax = lax.reduce(a, _constant_like(a, -np.inf), lax.max, dims)
   amax = lax.stop_gradient(lax.select(lax.is_finite(amax), amax, lax.full_like(amax, 0)))
   amax_singletons = dimadd(amax)
-  out = lax.add(lax.log(lax.reduce(lax.exp(lax.sub(a, amax_singletons)),
-                                   _constant_like(a, 0), lax.add, dims)), amax)
+  if b is None:
+    out = lax.add(lax.log(lax.reduce(lax.exp(lax.sub(a, amax_singletons)),
+                                     _constant_like(a, 0), lax.add, dims)), amax)
+    sign = jnp.ones_like(out)
+  else:
+    sumexp = lax.reduce(lax.mul(lax.exp(lax.sub(a, amax_singletons)), b),
```

Comment:
I believe `lax.sub` will fail here if `a` and `b` are different rank. We'll need to explicitly broadcast them first.

Reply:
So this should be able to work whether it's `b` that needs to be broadcast to the shape of `a` or vice versa. If I know which one needs to be broadcast, I can just use `jnp.broadcast_to`. But what if I don't know which one to broadcast?

Reply:
Done